### PR TITLE
feat(tv): Add TV plugin (@config-plugins/tv)

### PR DESCRIPTION
**Pull request by douglowder**
# Why

We need a way to allow Expo users to modify their Android and iOS projects to support TV.

After internal discussion, we intend to support that with this plugin, which will allow faster iteration than if these features are implemented within the CLI or elsewhere in the Expo monorepo.

# How

The plugin has an `isTV` setting, that can be enabled either with the environment (`EXPO_TV`) or an optional plugin parameter `isTV`.

When `isTV` is true, the iOS Podfile, Xcode project, and splash screen storyboard are modified to support TV.

When `isTV` is false, the plugin sees if the above TV modifications have been made, and reverts them.

The plugin requires (but does not check) that the React Native TV repo be used in place of the normal React Native dependency:

```json
{
  "dependencies": {
    "react-native": "npm:react-native-tvos@^0.72.4-0"
  }
}
```

If this is not the case, the plugin will run successfully, but Cocoapods installation will fail, since the React Native core repo does not support Apple TV.

# Test Plan

- Tested in a real project that produces EAS builds for phone and TV
- TBD: unit tests

### CODE REVIEW

**Comment by douglowder** [comment]
ENG-10042

**Comment by douglowder** [comment]
Test app to try this plugin out: https://github.com/douglowder/TestTV

File: packages/tv/src/withTVSplashScreen.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+) => {
+  const isTV = isTVEnabled(params);
+  const verbose = showVerboseWarnings(params);
+
+  return withDangerousMod(config, [
+    "ios",
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    async (config) => {
+      if (!config.modRequest.projectName) {
+        throw new Error("No project name");
```
Comment (byCedric) [inline packages/tv/src/withTVSplashScreen.ts:27]:
```suggestion
        throw new Error("The `@config-plugins/tv` plugin requires a configured project name.");
```

File: packages/tv/src/withTVXcodeProject.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+export function getIsTabletOnly(config: Pick<ExpoConfig, "ios">): boolean {
+  return !!config?.ios?.isTabletOnly;
+}
+
+export function getDeviceFamilies(config: Pick<ExpoConfig, "ios">): number[] {
+  const supportsTablet = getSupportsTablet(config);
+  const isTabletOnly = getIsTabletOnly(config);
+
+  if (isTabletOnly && config.ios?.supportsTablet === false) {
+    // add warning
```
Comment (byCedric) [inline packages/tv/src/withTVXcodeProject.ts:102]: 😄

Comment (douglowder) [reply packages/tv/src/withTVXcodeProject.ts:102]: I cut and pasted this from `@expo/config-plugins` 😄

File: packages/tv/src/withTVPodfile.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  if (src.indexOf("platform :tvos") === -1) {
+    return src;
+  }
+  const intermediateSrc = src.replace("platform :tvos", "platform :ios");
+  const newSrc = removeContents({
+    src: intermediateSrc,
+    tag: MOD_TAG,
+  }).contents;
+  if (!newSrc) {
+    throw new Error();
```
Comment (douglowder) [inline packages/tv/src/withTVPodfile.ts:63]:
```suggestion
    throw new Error(`${pkg.name}@${pkg.version}: Error in removing TV modifications from Podfile. Recommend running "npx expo prebuild --clean"`);
```

File: packages/tv/src/withTVSplashScreen.ts (hunk 1 above)
Comment (douglowder) [inline packages/tv/src/withTVSplashScreen.ts:27]:
```suggestion
        throw new Error(`The ${pkg.name}@${pkg.version} plugin requires a configured project name.`);
```

**Pull request merged**